barrels/LogMonkey/README.md

```diff
@@ -0,0 +1,77 @@
+| API             | Purpose                                                |
+| --------------- | ------------------------------------------------------ |
+| LogMonkey.Debug | General debugging information                          |
+| LogMonkey.Warn  | Warning messages about potential issues within the app |
+| LogMonkey.Error | Errors that have occurred within the app               |
+
+## Adding LogMonkey to Your Project
+
+See the general instructions for [including Monkey Barrels](https://github.com/garmin/connectiq-apps/tree/master/barrels#including-monkey-barrels).
+
+## Logging Information
+
+Once you've added LogMonkey to your project you can import it into your code via `using` statements.
+```
+using LogMonkey as Log;
+```
+Within you're project's code you can log messages, variables and Exceptions via the available APIs.
+```
+using LogMonkey as Log;
+
+class MyClassOfThings {
+
+    function aFunctionOfLogic(aParameter) {
+        Log.Debug.logMessage("MyClassOfThings", "aFunctionOfLogic()");
+        Log.Debug.logVariable("aFunctionOfLogic() param", "aParameter", aParameter);
+        ...
+        try {
+            ...do something horribly, horribly wrong...
+        } catch (ex) {
+            Log.Error.logException("Attempting to do XYZ", ex);
+        }
+        ...
+        if (aLocalVariable == aPotentiallyProblematicValue) {
+            Log.Warn.logVariable("aFunctionOfLogic()", "aLocalVariable", aLocalVariable);
+        }
+        ...
+    }
+
+}
+```
+
+## Viewing and Parsing Logs
+
+LogMonkey makes use of `Toybox.System.println()` calls to log information. This means that if your app also makes use of `pinrtln()` calls they will show up alongside the log entries.
+
+### Output Format
+
+LogMonkey logs each line in the file in the following format: `(log format version)[timestamp] {log level} tag: message`. Details for each block of this format are given in the table below.
+| Block              | Format                                                 | Description                                                                         |
+| ------------------ | ------------------------------------------------------ | ----------------------------------------------------------------------------------- |
+| log format version | A value beginning with "lfm" then a number             | The output format version of the line in the log.                                   |
+| timestamp          | MM-DD-YYYY HH:MM:SS                                    | A time stamp of when the log entry occurred.                                        |
+| log level          | Regular Expression: `[a-zA-Z0-9_]`                     | The priority level of the log entry. LogMonkey supports values of `D`, `W` and `E`. |
+| tag                | A string containing any character except a colon (`:`) | A brief tag value that can be used to quickly filter and find specific log entries. |
```

Comment:
The log level is described above using a regular expression, but this one in plain English. Should we be consistent?

Reply:
I think I choose to use a regular expression above because it was easier - from my view - to understand it but I'm going to use plain English for both.